src/ops/logical.rs

```diff
@@ -0,0 +1,1323 @@
+            return Err(DataStoreError::BroadcastError);
+        }
+
+        Ok(unsafe { self.logical_or_device_unchecked(other, stream) })
+    }
+
+    /// Approximate comparison of two arrays.
+    ///
+    /// The arrays are considered equal if:
+    ///
+    /// ```text
+    /// all(abs(a - b) <= (atol + rtol * abs(b)))
+    /// ```
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// use num_traits::Pow;
+    /// use mlx::Array;
+    /// let a = Array::from_slice(&[0., 1., 2., 3.], &[4]).sqrt();
+    /// let b = Array::from_slice(&[0., 1., 2., 3.], &[4]).pow(&(0.5.into()));
+    /// let mut c = a.all_close(&b, None, None, None);
+    ///
+    /// c.eval();
+    /// let c_data: &[bool] = c.as_slice();
+    /// // c_data == [true]
+    /// ```
+    ///
+    /// # Params
+    ///
+    /// - other: array to compare
+    /// - rtol: relative tolerance = defaults to 1e-5 when None
+    /// - atol: absolute tolerance - defaults to 1e-8 when None
+    /// - equal_nan: whether to consider NaNs equal -- default is false when None
+    /// - stream: stream or device to evaluate on
+    #[default_device]
+    pub fn all_close_device(
+        &self,
+        other: &Array,
+        rtol: impl Into<Option<f64>>,
+        atol: impl Into<Option<f64>>,
+        equal_nan: impl Into<Option<bool>>,
+        stream: StreamOrDevice,
+    ) -> Array {
+        self.try_all_close_device(other, rtol, atol, equal_nan, stream)
+            .unwrap()
+    }
+
+    /// Approximate comparison of two arrays without validating inputs.
+    ///
+    /// The arrays are considered equal if:
+    ///
+    /// ```text
+    /// all(abs(a - b) <= (atol + rtol * abs(b)))
+    /// ```
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// use num_traits::Pow;
+    /// use mlx::Array;
+    /// let a = Array::from_slice(&[0., 1., 2., 3.], &[4]).sqrt();
+    /// let b = Array::from_slice(&[0., 1., 2., 3.], &[4]).pow(&(0.5.into()));
+    /// let mut c = unsafe { a.all_close_unchecked(&b, None, None, None) };
+    ///
+    /// c.eval();
+    /// let c_data: &[bool] = c.as_slice();
+    /// // c_data == [true]
+    /// ```
+    ///
+    /// # Params
+    ///
+    /// - other: array to compare
+    /// - rtol: relative tolerance = defaults to 1e-5 when None
+    /// - atol: absolute tolerance - defaults to 1e-8 when None
+    /// - equal_nan: whether to consider NaNs equal -- default is false when None
+    /// - stream: stream or device to evaluate on
+    ///
+    /// # Safety
+    ///
+    /// This function is unsafe because it does not validate inputs.
+    #[default_device]
+    pub unsafe fn all_close_device_unchecked(
+        &self,
+        other: &Array,
+        rtol: impl Into<Option<f64>>,
+        atol: impl Into<Option<f64>>,
+        equal_nan: impl Into<Option<bool>>,
+        stream: StreamOrDevice,
+    ) -> Array {
+        let is_close = self.is_close_device_unchecked(other, rtol, atol, equal_nan, stream.clone());
```

Comment:
Why not just use `mlx_allclose`? Though these should be the same

Reply:
I guess for `unchecked` I could just use `mlx` directly. It shouldn't matter, but I can do that instead.